src/components/carousel/MainCarousel.jsx

```diff
@@ -3,28 +3,68 @@ import { useEffect, useState } from 'react';
 import CarouselCards from './shared-components/CarouselCards';
 import CarouselTextbox from './shared-components/CarouselTextbox';
 
+
+
 const MainCarousel = ({ images }) => {
   const [current, setCurrent] = useState(0);
   const [autoPlay, setAutoPlay] = useState(true);
+  
 
   let timeOut = null;
 
   const paginationDots = images.map((img, index) => <div className={index === current ? 'pagination_dot pagination_dot-active' : 'pagination_dot'} onClick={() => setCurrent(index) && clearTimeout(timeOut)}></div>);
 
   useEffect(
     () => {
-      timeOut =
+     let timeOut =
         autoPlay &&
         setTimeout(() => {
           slideRight();
         }, 5000);
+        return () => clearTimeout(timeOut);
     }
-    //timer function- disables error as timer doesn't need to be stored
   );
 
   const slideRight = () => {
     setCurrent(current === images.length - 1 ? 0 : current + 1);
-  }; // slide right and current checks what image it is on - it will go back to one if it reaches the end
+  }; 
+
+  const slideLeft = () => {
+    setCurrent(current === 0 ? images.length - 1 : current - 1);
+  }; //slide left and right functions - right is used for both the timer and touch events while left is only for touch events
+
+  const [touchStart, setTouchStart] = useState(null)
```

Comment:
UseStates and constants should be defined at the top of the function so we know what variables have been set as soon as we start reading the code

Structure is generally 

- UseStates
- UseEffects
- Variables
- Methods
- Return

Reply:
cleaned code and structured in this fashion